src/engine/renderer/tr_shader.cpp

```diff
@@ -3678,6 +3701,46 @@ static bool ParseShader( const char *_text )
 		}
 	}
 
+	// DarkPlaces compatibility
+	// It looks for extra maps based on their suffixes
+	int c;
+	shaderStage_t colorMapStage;
+
+	for ( c = 0; c < s; c++ ) {
+		if ( stages[c].type == stageType_t::ST_COLORMAP && stages[c].textureName != nullptr ) {
+			colorMapStage = stages[c];
+			break;
+		}
+	}
+
+	if ( colorMapStage.textureName[0] != '\0' )
+	{
+		const char *prefix; // not used but required by R_FindAltImage
+		const char *mapName;
+		char baseName[ MAX_QPATH ];
+
+		Log::Debug( "looking for extra maps for color map: '%s'", colorMapStage.textureName );
+		COM_StripExtension3( colorMapStage.textureName, baseName, MAX_QPATH );
+
+		int i;
+		for ( i = 0; i < numExtraMapParsers; i++ )
+		{
+			if ( s >= ( MAX_SHADER_STAGES - 1 ) )
+			{
+				Log::Warn( "too many extra stages in shader %s", shader.name );
+				return false;
+			}
+
+			mapName = va( "%s%s", baseName, extraMapParsers[i].suffix );
+			if( R_FindAltImage( (char*) mapName, &prefix ) >= 0 ) {
+				Log::Debug( "found extra %s '%s'", extraMapParsers[i].description, mapName );
+				// colorMapStage.type = stageType_t::ST_DIFFUSEMAP;
+				extraMapParsers[i].parser( &stages[ s ], &mapName );
```

Comment:
It looks to me like the 2nd parameter to Parse*Map functions is supposed the code of a q3 shader, rather than a filename.

Reply:
I've put `Log::Notice` everywhere, and when there is `normalMap something` in a shader, the original code passes ` something` to `ParseNormalMap`.

Reply:
for reference, this comment is misleading (I verified the keyword is never there):

https://github.com/DaemonEngine/Daemon/blob/caf9df48ec68021e7ebbf66d68972a14f11d6416/src/engine/renderer/tr_shader.cpp#L1330-L1336

this code says the keyword has no parameter, not that the keyword is not passed, which is the observed behavior:

https://github.com/DaemonEngine/Daemon/blob/caf9df48ec68021e7ebbf66d68972a14f11d6416/src/engine/renderer/tr_shader.cpp#L1352-L1356